lib/charms/layer/jenkins/plugins.py

```diff
@@ -19,21 +17,18 @@ def install(self, plugins):
 
         @params plugins: A whitespace-separated list of plugins to install.
         """
+        hookenv.log("Starting plugins installation process")
         plugins = plugins or ""
         plugins = plugins.split()
-        hookenv.log("Stopping jenkins for plugin update(s)")
-        host.service_stop("jenkins")
-        hookenv.log("Installing plugins (%s)" % " ".join(plugins))
+        plugins = (self._get_plugins_to_install(plugins))
 
         host.mkdir(
             paths.PLUGINS, owner="jenkins", group="jenkins", perms=0o0755)
-
-        existing_plugins = set(glob.glob("%s/*.hpi" % paths.PLUGINS))
+        existing_plugins = set(glob.glob("%s/*.jpi" % paths.PLUGINS))
         try:
             installed_plugins = self._install_plugins(plugins)
         except:
```

Comment:
every bare except is a bug, as it catches everything including exceptions that should never be caught (such as exit signals ala StopException, or memory errors). Catch a more specific error, or worst case 'except Exception:'. But generally even that is a bug waiting to happen.

Reply:
This bare except is only used to log a message and then the exception is re-raised.
```suggestion
        except:
            hookenv.log("Plugin installation failed, check logs for details")
            raise
```

Reply:
This bare except is only used to log a message and then the exception is re-raised:
```
        except:
            hookenv.log("Plugin installation failed, check logs for details")
            raise
```

Reply:
I changed it to ```except Exception:``` because I believe it addresses the same need.